pkg/kubectl/cmd/cmd.go

```diff
@@ -169,30 +169,41 @@ __custom_func() {
 	// and add a short forms entry in expandResourceShortcut() when appropriate.
 	// TODO: This should be populated using the discovery information from apiserver.
 	valid_resources = `Valid resource types include:
+   * certificatesigningrequests (aka 'csr')
+   * clusterrolebindings
+   * clusterroles
    * clusters (valid only for federation apiservers)
    * componentstatuses (aka 'cs')
    * configmaps (aka 'cm')
    * daemonsets (aka 'ds')
    * deployments (aka 'deploy')
-   * events (aka 'ev')
    * endpoints (aka 'ep')
+   * events (aka 'ev')
    * horizontalpodautoscalers (aka 'hpa')
-   * ingress (aka 'ing')
+   * ingresses (aka 'ing')
    * jobs
    * limitranges (aka 'limits')
-   * nodes (aka 'no')
    * namespaces (aka 'ns')
+   * networkpolicies
+   * nodes (aka 'no')
+   * persistentvolumeclaims (aka 'pvc')
+   * persistentvolumes (aka 'pv')
    * petsets (alpha feature, may be unstable)
+   * podSecurityPolicies (aka 'psp')
    * pods (aka 'po')
-   * persistentvolumes (aka 'pv')
-   * persistentvolumeclaims (aka 'pvc')
-   * quota
-   * resourcequotas (aka 'quota')
+   * podtemplates
    * replicasets (aka 'rs')
    * replicationcontrollers (aka 'rc')
+   * resourcequotas (aka 'quota')
+   * rolebindings
+   * roles
+   * scheduledjobs
    * secrets
    * serviceaccounts (aka 'sa')
    * services (aka 'svc')
+   * storageclasses
+   * thirdpartyresourcedatas
```

Comment:
I didn't knew about datum/data.

I removed it as it's not in the [list](https://github.com/ymqytw/misc/blob/master/resources.json) from @ymqytw.

I found this type originally in the bash completion suggestions (as generated by `kubectl completion bash`).
The singular name is generated [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/resource_printer.go#L441) from this type `*extensions.ThirdPartyResourceData` and the pluralization code is [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/kubectl.go#L142)  so thirdtpartyresourcedatas would be the generated "plural".